packages/recommend-core/src/personalization/getAffinities.ts

```diff
@@ -0,0 +1,41 @@
+type GetAffinities = {
+  userToken: string;
+  region: string;
+  apiKey: string;
+  appId: string;
+};
+
+type AffinitiesResponse = {
+  userToken: string;
+  lastEventAt: string;
+  scores: Record<string, Record<string, number>>;
+};
+
+export const getAffinities = async ({
+  userToken,
+  region,
+  apiKey,
+  appId,
+}: GetAffinities): Promise<AffinitiesResponse> => {
+  const response = await fetch(
+    `https://personalization.${region}.algolia.com/1/profiles/personalization/${encodeURIComponent(
+      userToken
+    )}`,
+    {
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Algolia-Application-Id': appId,
+        'X-Algolia-API-Key': apiKey,
+      },
+    }
+  );
+
+  if (!response.ok) {
+    throw new Error(
+      `Failed to fetch personalisation affinities. Status: ${response.status}`
+    );
+  }
```

Comment:
Do we really want to break the recommendations behavior on the end-user screen when we're unable to fetch their user profile?

We better fallback to non-personalized recommendations. That's similar to how the Search Engine deals with non-existant user tokens, and that's similar to the future Recommend backend behavior.

(Same comment for the strategy fetching.)

Reply:
There is a `try/catch` around these calls in `getPersonalizationFilters`. 

This throw here, is to check the HTTP status since fetch won't throw on `4xx` errors.

TL;DR when a userToken results in an HTTP error we fallback silently to non-personalised recommend.

Reply:
Indeed, I missed that